pkg/steps/artifacts.go

```diff
@@ -169,16 +176,23 @@ type PodClient interface {
 	WithNewLoggingClient() PodClient
 	Exec(namespace, pod string, opts *coreapi.PodExecOptions) (remotecommand.Executor, error)
 	GetLogs(namespace, name string, opts *coreapi.PodLogOptions) *rest.Request
+	RealClient() kubernetes.Interface
 }
 
 func NewPodClient(ctrlclient loggingclient.LoggingClient, config *rest.Config, client rest.Interface) PodClient {
-	return &podClient{LoggingClient: ctrlclient, config: config, client: client}
+	return &podClient{
```

Comment:
This entirely breaks the loggingclient abstraction for pods

Reply:
> This entirely breaks the loggingclient abstraction for pods

I doubt that it breaks a logging abstraction since the logging client was not touched.

Reply:
That's the problem: https://github.com/openshift/ci-tools/blob/252a894afa981c86712afdf0bc826adc73cf5d81/pkg/steps/loggingclient/loggingclient.go#L43

Reply:
Thanks.  Added a commit to do a single GET to keep the cache as up to date as possible for future usage.